onnxruntime/core/providers/cuda/math/gemm.cc

```diff
@@ -123,21 +129,38 @@ Status Gemm<T>::ComputeInternal(OpKernelContext* ctx) const {
 
   CudaT alpha = ToCudaType<T>::FromFloat(alpha_);
   CudaT beta = ToCudaType<T>::FromFloat(beta_);
-  // Gemm, note that CUDA assumes col-major, so Y(N,M) = alpha * op(W) x op(X) + beta * Y
-  CUBLAS_RETURN_IF_ERROR(cublasGemmHelper(
-      CublasHandle(),
-      trans_B_ ? CUBLAS_OP_T : CUBLAS_OP_N,
-      trans_A_ ? CUBLAS_OP_T : CUBLAS_OP_N,
-      N, M, K,
-      &alpha,
-      reinterpret_cast<const CudaT*>(W->Data<T>()),
-      (trans_B_ ? K : N),
-      reinterpret_cast<const CudaT*>(X->Data<T>()),
-      (trans_A_ ? M : K),
-      // ideally we need to set the output buffer contents to 0 if bias is missing,
-      // but passing 0 for beta is cheaper and it will ignore any junk in the output buffer
-      B != nullptr ? &beta : &zero,
-      out_data, N, device_prop));
+
+  // General note : CUDA assumes col-major, so Y(N,M) = alpha * op(W) x op(X) + beta * Y
+  if (use_cublaslt_matmul_) { // Use CublasLtMatmul
+    CUBLAS_RETURN_IF_ERROR(cublasLtMatmulHelper(
+        CublasLtHandle(),
+        trans_B_ ? CUBLAS_OP_T : CUBLAS_OP_N,
+        trans_A_ ? CUBLAS_OP_T : CUBLAS_OP_N,
+        N, M, K,
+        &alpha,
+        reinterpret_cast<const CudaT*>(W->Data<T>()),
+        (trans_B_ ? K : N),
+        reinterpret_cast<const CudaT*>(X->Data<T>()),
+        (trans_A_ ? M : K),
+        (B != nullptr) ? &beta : &zero,
```

Comment:
We still use the same concept as for CublasGemm while dealing with bias - we copy it over to the output buffer before invoking cublasLtMatmul. There is another way to do this - using CublasLtMatmul epilogue bias. But scaling of epilogue bias isn't support (i.e.) beta cannot be honored. So the way to use epilogue bias here is actually to scale the bias with beta before passing it onto CublasLtMatmul. Not sure if that will be faster than this approach of copying the bias to the output buffer and just using:
 `output_buffer = alpha * A * B + beta * output_buffer`

Will defer using epligoue bias until it is proven that it is needed